oci/spec_opts_test.go

```diff
@@ -0,0 +1,84 @@
+package oci
+
+import (
+	"context"
+	"testing"
+
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/namespaces"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func TestSpecWithEnvAppend(t *testing.T) {
+	t.Parallel()
+
+	ctx := namespaces.WithNamespace(context.Background(), "testing")
+	s, err := GenerateSpec(ctx, nil, &containers.Container{ID: t.Name()}, WithEnv([]string{
```

Comment:
I think you can test `WithEnv` by calling it directly, instead of calling `GenerateSpec`.